web/src/data/recoil.ts

```diff
@@ -110,12 +110,38 @@ export const vaporConnectionConfig = selector<ConnectionConfig | undefined>({
   },
 });
 
+export const portalConnectionConfig = selector<ConnectionConfig | undefined>({
+  key: "portalConnectionConfig",
+  get: async ({ get }) => {
+    const {search} = window.location;
+    if (search) {
+      const queryParams = new URLSearchParams(search);
+      const portalHostname = queryParams.get("hostname");
+      const portalCredentials = queryParams.get("credentials");
```

Comment:
use atoms to access query params - similar to how vapor does it above

Reply:
oh, I didn't notice that. Thanks